kong/db/dao/workspaces.lua

```diff
@@ -18,4 +25,14 @@ function Workspaces:truncate()
 end
 
 
+function Workspaces:select_by_name(key, options)
+  if kong.configuration.database == "off" and key == "default" then
+    -- it should be a table, not a single string
+    return  { id = lmdb.get(DECLARATIVE_DEFAULT_WORKSPACE_KEY), }
+  end
+
+  return self.super.select_by_name(self, key, options)
+end
```

Comment:
if we remove this new function, the following failed test case will pass, which is only reproduced by CI so that i cannot reproduce it locally on my macOS.

```
       -------
        failed   #wasm wasmtime (role: traditional) kong stop/start/restart does not introduce any errors
        FAIL spec/02-integration/20-wasm/10-wasmtime_spec.lua:138:
        __________
spec/02-integration/20-wasm/10-wasmtime_spec.lua:163: 2024/10/08 09:00:08 [warn] the configuration value 'strict' for configuration property 'worker_consistency' is deprecated
2024/10/08 09:00:08 [warn] the 'worker_consistency' configuration property is deprecated
2024/10/08 09:00:08 [warn] the 'opentelemetry_tracing' configuration property is deprecated, use 'tracing_instrumentations' instead
2024/10/08 09:00:08 [warn] the 'opentelemetry_tracing_sampling_rate' configuration property is deprecated, use 'tracing_sampling_rate' instead
Error: ./kong/cmd/start.lua:99: nginx: [error] [lua] init.lua:743: init(): kong.default_workspace: 1:nil
Error: nginx: [error] [lua] init.lua:745: init(): kong.default_workspace: 2:nil
Error: nginx: [error] init_by_lua error: ./kong/runloop/plugins_iterator.lua:498: table index is nil
stack traceback:
        ./kong/runloop/plugins_iterator.lua:498: in function 'PluginsIterator_new'
        ./kong/runloop/handler.lua:512: in function 'new_plugins_iterator'
        ./kong/runloop/handler.lua:528: in function 'build_plugins_iterator'
        ./kong/init.lua:747: in function 'init'
        init_by_lua(nginx-kong.conf:46):3: in main chunk
```

Reply:
Both the busted test framework and this function has bugs, so here are two pr to fix them:

1. https://github.com/Kong/kong/pull/13738/
2. https://github.com/Kong/kong/pull/13733